### PR TITLE
Update for Python 3

### PR DESCRIPTION
**Pull request by mrinehart-r7**
Updates the codebase for Python 3 compatibility

### CODE REVIEW

File: src/scripts/pymsfconsole.bak [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (mberezin-r7) [reply src/scripts/pymsfconsole.bak]: Files excluded using `bfg --delete-files pymsfconsole.bak .`

File: src/scripts/pymsfrpc.bak [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (mberezin-r7) [reply src/scripts/pymsfrpc.bak]: Files excluded using `bfg --delete-files pymsfrpc.bak .`

File: src/metasploit/msfrpc.py [hunk 3]
```diff
@@ -1461,14 +1461,15 @@ def execute(self, **kwargs):
                             'Invalid payload (%s) for given target (%d).' % (payload.modulename, self.target)
                         )
                     runopts['PAYLOAD'] = payload.modulename
-                    for k, v in payload.runoptions.iteritems():
-                        if v is None or (isinstance(v, basestring) and not v):
+
+                    for k, v in list(payload.runoptions.items()):
```
Comment (mrinehart-r7) [inline src/metasploit/msfrpc.py:1465]: Is the cast to a `list` necessary? `.items()` already provides iteration with key & value - casting to a list just creates a list of tuples that doesn't seem to provide any additional benefit.

Comment (mberezin-r7) [reply src/metasploit/msfrpc.py:1465]: Indeed this appears unnecessary, removing.

```python
>>> test = {'a': 1, 'b': 2, 'c': 3}
>>> test
{'a': 1, 'b': 2, 'c': 3}
>>> type(test)
<class 'dict'>
>>> for k,v in test.items():
...     print(f'k is {type(k)} v is {type(v)}')
...
...
k is <class 'str'> v is <class 'int'>
k is <class 'str'> v is <class 'int'>
k is <class 'str'> v is <class 'int'>
```

Comment (allfro) [reply src/metasploit/msfrpc.py]: Yes, it is absolutely redundant. Poor code on my part :)

**Review by allfro — COMMENTED** [review]
Could we please remove `.idea` from the commit history. It's not required and keeps the project IDE agnostic. If you look at the commit contents you can see that references to `komand-pymetasploit` have been introduced. The gitignore entry for `.idea/` as a whole should have been left in.

**Comment by mberezin-r7** [comment]
> Could we please remove `.idea` from the commit history. It's not required and keeps the project IDE agnostic. If you look at the commit contents you can see that references to `komand-pymetasploit` have been introduced. The gitignore entry for `.idea/` as a whole should have been left in.

Sure thing, just a suggestion but not married to the idea (pun intended). Removing JetBrains references.

**Comment by allfro** [comment]
Thanks for the awesome PR :)

**Pull request merged**